### ISSUE TITLE
Cannot catkin_make

### ISSUE THREAD

**Issue by Kn15263ss** [ThundeRatz/ros_yolo2#10]
Hi~ I have a little problem..
When i catkin_make, it encounter a problem:
![image](https://user-images.githubusercontent.com/22387638/29179957-c187feb4-7e28-11e7-8e56-7b3359c913e5.png)
![image](https://user-images.githubusercontent.com/22387638/29179972-cf6993da-7e28-11e7-9cbf-d19a57eef49c.png)
![image](https://user-images.githubusercontent.com/22387638/29179980-d6709e58-7e28-11e7-9274-8947fd0e3975.png)

How do i fix this problem?
Thanks you!!

**Comment by tiagoshibata** [comment]
Hi @Kn15263ss,

You seem to be using a too old GCC or CMake version. Our `CMakeLists.txt` sets `CMAKE_CXX_STANDARD 11` ([here](https://github.com/ThundeRatz/ros_yolo2/blob/c1a4994cfa6de7b943eae5ab795023eb46192725/CMakeLists.txt#L7)), which should 
pass the `-std=gnu++11` flag to g++ if using `cmake` >= 3.1. Updating CMake should make it recognize the variable and pass the `-std=c++11` flag on to GCC.

Furthermore, in GCC > 6, the C++14 standard is the default one and the flag shouldn't be needed.

Try updating CMake and/or GCC and recompile. If updating your toolchain is not possible, [this thread](https://stackoverflow.com/questions/10851247/how-to-activate-c-11-in-cmake) has a workaround to set the C++ standard on old versions of CMake.

Let me know if you need anything else!

**Comment by Kn15263ss** [comment]
Hi @tiagoshibata, thanks you!
I have update my GCC and CMake to 7 and 3.2.2
![image](https://user-images.githubusercontent.com/22387638/29208862-60574c3a-7ebe-11e7-9d5a-570eb06a73b5.png)

![image](https://user-images.githubusercontent.com/22387638/29208802-24a5fe20-7ebe-11e7-9c8a-cb81203bbb6b.png)

but i have encounter a another problem:
![image](https://user-images.githubusercontent.com/22387638/29208906-8ef4798c-7ebe-11e7-8152-5f21fed95a5b.png)

What happen about this?

Thanks you again!

**Comment by tiagoshibata** [comment]
Have you followed the instructions on [README.md](https://github.com/ThundeRatz/ros_yolo2/blob/master/README.md)?

> If building from source, change to the src directory of your workspace and use git clone --recursive https://github.com/tiagoshibata/ros_yolo2.git to download with submodules.

It seems like the darknet submodule wasn't cloned; try passing `--recursive` to `git clone`.

**Comment by Kn15263ss** [comment]
Thanks you very much!! It successful to run!

But, when I run it with webcam, and put on my own train xxx.cfg and xxx.weight file to data folder, then check the message from "rostopic echo /vision/yolo2/detections", then the result:
![image](https://user-images.githubusercontent.com/22387638/29219039-6c5d67dc-7ee9-11e7-9375-ed2127583e8f.png)
It seem cannot find any object.

I have test it on "./darknet detector demo data/obj.name cfg/yolo.cfg yolo.weight", it can do well.
so..I don't know why, 

Thanks you!

**Comment by tiagoshibata** [comment]
Are you training on YOLO9000 (the latest datknet version)? This repo uses an slightly older version of Darknet and some of the changes introduced with YOLO9000 might not work.

Try compiling the Darknet version used in this repository (submodule darknet) and running it. If it works on this Darknet version and fails in the nodelet, them it's a bug.

We have been using YOLO tiny successfully. The following configuration works with this Darknet version:

```
[net]
batch=64
subdivisions=16
width=416
height=416
channels=3
momentum=0.9
decay=0.0005
angle=0
saturation = 1.5
exposure = 1.5
hue=.1

learning_rate=0.001
max_batches = 40100
policy=steps
steps=-1,100,20000,30000
scales=.1,10,.1,.1

[convolutional]
batch_normalize=1
filters=16
size=3
stride=1
pad=1
activation=leaky

[maxpool]
size=2
stride=2

[convolutional]
batch_normalize=1
filters=32
size=3
stride=1
pad=1
activation=leaky

[maxpool]
size=2
stride=2

[convolutional]
batch_normalize=1
filters=64
size=3
stride=1
pad=1
activation=leaky

[maxpool]
size=2
stride=2

[convolutional]
batch_normalize=1
filters=128
size=3
stride=1
pad=1
activation=leaky

[maxpool]
size=2
stride=2

[convolutional]
batch_normalize=1
filters=256
size=3
stride=1
pad=1
activation=leaky

[maxpool]
size=2
stride=2

[convolutional]
batch_normalize=1
filters=512
size=3
stride=1
pad=1
activation=leaky

[maxpool]
size=2
stride=1

[convolutional]
batch_normalize=1
filters=1024
size=3
stride=1
pad=1
activation=leaky

###########

[convolutional]
batch_normalize=1
size=3
stride=1
pad=1
filters=1024
activation=leaky

[convolutional]
size=1
stride=1
pad=1
filters=30
activation=linear

[region]
anchors = 1.08,1.19,  3.42,4.41,  6.63,11.38,  9.42,5.11,  16.62,10.52
bias_match=1
classes=1
coords=4
num=5
softmax=1
jitter=.2
rescore=1

object_scale=5
noobject_scale=1
class_scale=1
coord_scale=1

absolute=1
thresh = .6
random=1
```

We'll probably update the darknet version sometime, we just didn't need any of the newest features.

**Comment by Kn15263ss** [comment]
I have re-train a weights by use yolo-ting.cfg(follow your example), then it still didn't detect any object.
![image](https://user-images.githubusercontent.com/22387638/29247679-b51f1cdc-803c-11e7-8fdf-9f3a5f236bf1.png)

**Comment by tiagoshibata** [comment]
You confidence level might be too high; try setting the node's `confidence` parameter to something lower. The default value is 0.8.

Just double checking, did you run it using the darknet version in this repository (the submodule in the darknet folder)?

**Comment by Kn15263ss** [comment]
I have two darknet folder, one is at my own path, the other is at your ros_yolo2 folder, and I test it(ex:./darknet detector demo xx xx) from my darknet.
And, the confidence parameter, it value is 0.5.

**Comment by tiagoshibata** [comment]
Try lowering the confidence parameter in the node. If you're running `./darknet` with 0.5 confidence and the node with the default value (0.8), that's probably the reason you aren't getting detections.

See [this page](http://wiki.ros.org/Nodes#Remapping_Arguments.Node_parameter_assignment) for node parameter assignment or the [README.md](https://github.com/ThundeRatz/ros_yolo2/blob/master/README.md) file if using a launchfile. Let me know if it solves your issue!

**Comment by Kn15263ss** [comment]
I have change the confidence from yolo2_nodelet.cpp to 0.5, and re-catkin_make.
But, It is still not detecting ant object.
![image](https://user-images.githubusercontent.com/22387638/29269880-7dae021e-8126-11e7-8f54-656c835ce3b6.png)
![image](https://user-images.githubusercontent.com/22387638/29269969-f8219bf0-8126-11e7-95b3-416ff986b633.png)

**Comment by tiagoshibata** [comment]
Hmm, that's weird... Which camera node/nodelet are you using? Are you sure the camera stream is correct? Have you tried the sample launchfile given in the README.md of this repo (using `uvc_camera`)?

**Comment by Kn15263ss** [comment]
I using uvc_camera/CameraNodelet to run it.

**Comment by tiagoshibata** [comment]
I can try to run it locally if you're willing to share your files, but it might take some time (I'm a bit busy at the moment).

**Comment by Kn15263ss** [comment]
It was push on google cloud: https://drive.google.com/open?id=0B3pT9uJW73uAb0MwVm0zYzBad1k
And very thank you!!

**Comment by Kn15263ss** [comment]
@tiagoshibata 
Hi~ I have solved the problem, that can detecting object!! 
But, I don't know how I solved it :D...
I just put my own data, label and config on your darknet from ros_yolo2, and train on it!
Finally, It can detecting object!
So..I don't know why I use my own darknet can't working, but use your can do it.

**Comment by tiagoshibata** [comment]
Great :) ! That's what I meant when I suggested using the Darknet version in the submodule of this repository. Darknet had breaking changes in the latest versions.
